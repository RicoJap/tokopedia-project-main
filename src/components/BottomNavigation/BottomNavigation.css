.BottomNavigation-wrapper {
    position: fixed;
    bottom: 0;
    width: 100%;
    /* background-color: #7D4CDB !important; */
}

/* .BottomNavigation-action {
    color: #fff !important;
} */