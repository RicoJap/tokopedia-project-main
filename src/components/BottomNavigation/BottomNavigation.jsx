import React from "react";
import MuiBottomNavigation from "@material-ui/core/BottomNavigation";
import MuiBottomNavigationAction from "@material-ui/core/BottomNavigationAction";
import { makeStyles } from "@material-ui/core/styles";

const useBottomNavigationStyles = makeStyles({
  root: {
		position: 'fixed',
    bottom: 0,
    width: '100%',
  },
});

const useBottomNavigationActionStyles = makeStyles({
  root: {
    "&$selected": {
      color: "#7D4CDB",
    },
  },
  selected: {
    color: "#7D4CDB",
  },
});

/**
 * BottomNavigation Component
 *
 * @since  16/11/2020
 * @access (public)
 *
 * @type     Functional Component
 * @param {Array}  actions  Actions inside the bottom navigation for users to select which action
 * @param {Number}  selectedAction  The selected action in the form of index of the actions array
 * @param {Function}  onChange  onChange handler
 *
 */

const BottomNavigation = ({ actions, selectedAction, onChange, ...props }) => {

	const bottomNavigationStyles = useBottomNavigationStyles();
	const bottomNavigationActionClasses = useBottomNavigationActionStyles();

  return (
    <MuiBottomNavigation
      value={selectedAction}
      onChange={onChange}
      showLabels
      classes={bottomNavigationStyles}
      {...props}
    >
      {actions.map((action, i) => {
        return (
          <MuiBottomNavigationAction
            classes={bottomNavigationActionClasses}
            key={i}
            label={action.label}
            icon={!!action.icon && action.icon}
            data-testid={action.label}
          />
        );
      })}
    </MuiBottomNavigation>
  );
};

export default BottomNavigation;
