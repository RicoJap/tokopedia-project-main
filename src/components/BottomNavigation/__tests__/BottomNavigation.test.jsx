import "@testing-library/jest-dom";
import React from "react";
import renderer from "react-test-renderer";

import { fireEvent, render, screen, waitFor } from "@testing-library/react";

import BottomNavigation from "../BottomNavigation";

const mockOnChange = jest.fn();
const mockSelectedAction = jest.mock();

const MockBottomNavigation = () => {
  return (
    <BottomNavigation
      selectedAction={mockSelectedAction}
      onChange={mockOnChange}
      actions={[
        { label: "Action 1" },
        { label: "Action 2"},
      ]}
    />
  );
};

describe('Snapshot testing', () => {
  const tree = renderer.create(<MockBottomNavigation />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('BottomNavigation Component renders correctly', () => {
  test('Check if BottomNavigation component is present in the document', () => {
		render(<MockBottomNavigation />);
		expect(screen.getByTestId('Action 1')).toBeInTheDocument();
		expect(screen.getByTestId('Action 2')).toBeInTheDocument();
  });
});

describe('Test BottomNavigation component event handler', () => {
  test('BottomNavigation component onChange() triggered when one of the action button is pressed', async () => {
    render(<MockBottomNavigation />);
    await waitFor(() => {
      fireEvent.click(screen.getByTestId('Action 1'));
    });

    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });
});
