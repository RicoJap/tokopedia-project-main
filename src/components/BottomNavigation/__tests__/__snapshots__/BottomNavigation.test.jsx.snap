// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="MuiBottomNavigation-root makeStyles-root-1"
>
  <button
    className="MuiButtonBase-root MuiBottomNavigationAction-root makeStyles-root-2"
    data-testid="Action 1"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiBottomNavigationAction-wrapper"
    >
      <span
        className="MuiBottomNavigationAction-label"
      >
        Action 1
      </span>
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiBottomNavigationAction-root makeStyles-root-2"
    data-testid="Action 2"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiBottomNavigationAction-wrapper"
    >
      <span
        className="MuiBottomNavigationAction-label"
      >
        Action 2
      </span>
    </span>
  </button>
</div>
`;
